FabNESO/tasks.py

```diff
@@ -90,16 +91,47 @@ def neso_ensemble(
         solver: Which NESO solver to use.
         conditions_file_name: Name of conditions XML file in configuration directory.
         mesh_file_name: Name of mesh XML in configuration directory.
+        parameter_scans is a set of parameters to sweep over. A slash separated list of
+        lower bound, upper bound, and steps.
     """
-    path_to_config = fab.find_config_file_path(config)
-    sweep_dir = str(Path(path_to_config) / "SWEEP")
-    fab.update_environment(
-        {
-            "script": "neso",
-            "neso_solver": solver,
-            "neso_conditions_file": conditions_file_name,
-            "neso_mesh_file": mesh_file_name,
-        }
+    path_to_config = Path(fab.find_config_file_path(config))
+    temporary_context: TemporaryDirectory | nullcontext = (
+        TemporaryDirectory(prefix=f"{config}_", dir=path_to_config.parent)
+        if parameter_scans != {}
+        else nullcontext()
     )
-    fab.with_config(config)
-    fab.run_ensemble(config, sweep_dir)
+    with temporary_context as temporary_config_directory:
+        if parameter_scans != {}:
+            temporary_config_path = Path(temporary_config_directory)
+            # Because FabSIM is a bit weird with commas, build the dict here
+            parameter_scan_dict = {
+                parameter: (
+                    float(values.split("/")[0]),
+                    float(values.split("/")[1]),
+                    int(values.split("/")[2]),
```

Comment:
I would factor out the parsing of the strings here to a separate function to avoid having to split multiple times, something like
```Python
def _parse_parameter_scan_string(
    parameter_scan_string: str, delimiter: str,
) -> tuple[float, float, int]:
    start, end, n_steps = parameter_scan_string.split(delimiter)
    return float(start), float(end), int(n_steps)
```

This should already raise a relatively informative error if the string is not of the expected format (as won't be able to unpack split string into three arguments) but could also explicitly check length of split string and raise an error if not equal to 3.